packages/wallet/src/Wallets/BaseWallet.ts

```diff
@@ -114,15 +117,82 @@ import isEqual from 'lodash/isEqual';
 import uniq from 'lodash/uniq';
 import type { KoraLabsHandleProvider } from '@cardano-sdk/cardano-services-client';
 
-export interface PersonalWalletProps {
+/**
+ * Method to derive the wallet's grouped address from the script and network ID.
+ *
+ * @param {Cardano.NativeScript} paymentScript - The native script to be used for the payment credential.
+ * @param {Cardano.NativeScript} stakingScript - The native script to be used for the staking credential.
+ * @param {Cardano.NetworkId} networkId - The network identifier.
+ * @returns {}GroupedAddress} The derived grouped address.
+ */
+const getScriptAddress = (
```

Comment:
Please hoist this to `util.ts`, this is already a large file

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1111/commits/cc00e21b8e3be0da410c52e9b310dbba096fcf07

Reply:
I had to move this function to its own file (internals.ts) because utils.ts has many utils that it exports and depends on the BaseWallet, so there is a cyclic dependency if we put this function there